akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -57,9 +83,7 @@ import akka.annotation.InternalApi
    */
   private var failures: Map[untyped.ActorRef, Throwable] = Map.empty
 
-  def receive: Receive = running
-
-  def running: Receive = {
+  protected final def receiveMessage(msg: Any): Unit = msg match {
     case untyped.Terminated(ref) =>
```

Comment:
we can try if it matters if we collapse these 4 cases into 2
Terminated and ReceiveTimeout are PossiblyHarmful
AdaptMessage and AdaptWithRegisteredMessageAdapter are InternalMessage

Reply:
yeah, I didn't because when I tried introducing a common `InternalMessage` for all of this it pretty consistently made performance worse for some reason.